src/bindgen/ir/enumeration.rs

```diff
@@ -700,28 +700,24 @@ impl Source for Enum {
                         for &(ref field_name, ref ty, ..) in body.fields.iter().skip(skip_fields) {
                             out.new_line();
                             match ty {
-                                Type::Array(ref _ty, ref length) => {
+                                Type::Array(ref ty, ref length) => {
                                     // arrays are not assignable in C++ so we
                                     // need to manually copy the elements
+                                    write!(out, "for (int i = 0; i < {}; i++)", length.as_str());
+                                    out.open_brace();
                                     write!(
                                         out,
-                                        "for (int i = 0; i < {}; i++) {{\
-                                         result.{}.{}[i] = {}[i];\
-                                         }}",
-                                        length.as_str(),
-                                        variant_name,
-                                        field_name,
-                                        arg_renamer(field_name)
+                                        "new (result.{}.{} + i) ",
+                                        variant_name, field_name
                                     );
+                                    ty.write(config, out);
```

Comment:
Writing out the bare type in a new expression usually works out, but with some more complex types like function pointers, it's generally a better idea to write out the type itself in parentheses:

```c++
::new (&result.foo.x[i]) (int32_t)();
// -or, with a function pointer-
::new (&result.foo.x[i]) (void (*) (int32_t)) (); // max readable
```

Reply:
Good point